cocotb/share/lib/simulator/simulatormodule.cpp

```diff
@@ -926,133 +926,152 @@ static PyMethodDef gpi_sim_hdl_methods[] = {
         (PyCFunction)get_signal_val_long, METH_NOARGS, PyDoc_STR(
             "get_signal_val_long($self)\n"
             "--\n\n"
-            "Get the value of a signal as a long"
+            "get_signal_val_long(self) -> int\n"
+            "Get the value of a signal as an integer."
         )
     },
     {"get_signal_val_str",
         (PyCFunction)get_signal_val_str, METH_NOARGS, PyDoc_STR(
             "get_signal_val_str($self)\n"
             "--\n\n"
-            "Get the value of a signal as an ASCII string"
+            "get_signal_val_str(self) -> str\n"
+            "Get the value of a signal as a string."
         )
     },
     {"get_signal_val_binstr",
         (PyCFunction)get_signal_val_binstr, METH_NOARGS, PyDoc_STR(
             "get_signal_val_binstr($self)\n"
             "--\n\n"
-            "Get the value of a signal as a binary string"
+            "get_signal_val_binstr(self) -> str\n"
+            "Get the value of a logic vector signal as a string of (``0``, ``1``, ``X``, etc.), one element per character."
         )
     },
     {"get_signal_val_real",
         (PyCFunction)get_signal_val_real, METH_NOARGS, PyDoc_STR(
             "get_signal_val_real($self)\n"
             "--\n\n"
-            "Get the value of a signal as a double precision float"
+            "get_signal_val_real(self) -> float\n"
+            "Get the value of a signal as a float."
         )
     },
     {"set_signal_val_long",
         (PyCFunction)set_signal_val_long, METH_VARARGS, PyDoc_STR(
             "set_signal_val_long($self, action, value, /)\n"
             "--\n\n"
-            "Set the value of a signal using a long"
+            "set_signal_val_long(self, action: int, value: int) -> None\n"
+            "Set the value of a signal using an integer.\n"
         )
     },
     {"set_signal_val_str",
         (PyCFunction)set_signal_val_str, METH_VARARGS, PyDoc_STR(
             "set_signal_val_str($self, action, value, /)\n"
             "--\n\n"
-            "Set the value of a signal using an NUL-terminated 8-bit string"
+            "set_signal_val_str(self, action: int, value: str) -> None\n"
+            "Set the value of a signal using a user-encoded string."
         )
     },
     {"set_signal_val_binstr",
         (PyCFunction)set_signal_val_binstr, METH_VARARGS, PyDoc_STR(
             "set_signal_val_binstr($self, action, value, /)\n"
             "--\n\n"
-            "Set the value of a signal using a string with a character per bit"
+            "set_signal_val_binstr(self, action: int, value: str) -> None\n"
+            "Set the value of a logic vector signal with a string of (``0``, ``1``, ``X``, etc.), one element per character."
```

Comment:
```suggestion
            "Set the value of a logic vector signal using a string of (``0``, ``1``, ``X``, etc.), one element per character."
```
For consistency with the other strings, or does it roll better with ``with``?